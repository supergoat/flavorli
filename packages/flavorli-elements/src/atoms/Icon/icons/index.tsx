import cookingTime from './cooking_time.svg';
import difficulty from './difficulty.svg';
import ingredients from './ingredients.svg';
import preparationTime from './preparation_time.svg';
import preparation from './preparation.svg';
import serves from './serves.svg';
import timer from './timer.svg';
import boiling from './boiling.svg';

export const icons = {
  cookingTime,
  difficulty,
  ingredients,
  preparationTime,
  preparation,
  serves,
  timer,
  boiling,
};
