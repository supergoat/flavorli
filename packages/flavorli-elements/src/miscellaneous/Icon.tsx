import React from 'react';
import styled from 'styled-components';
import {svg} from '../images';

export type IconName = keyof typeof svg;
export interface IconProps {
  name: IconName;
  size?: string;
  className?: string;
}
export const Icon = ({name, size, className}: IconProps) => {
  const icon = (svg[name] as unknown) as string;
  return <IconWrapper src={icon} size={size} className={className} alt="" />;
};

const IconWrapper = styled.img<{size?: string}>`
  width: ${props => props.size};
  height: ${props => props.size};
`;
