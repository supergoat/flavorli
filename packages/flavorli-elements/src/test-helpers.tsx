import React from 'react';
import {ThemeProvider} from 'styled-components';
import {render as testRender} from '@testing-library/react';
import {theme} from '.';

export function render(ui: React.ReactNode) {
  const utils = testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

  return utils;
}
