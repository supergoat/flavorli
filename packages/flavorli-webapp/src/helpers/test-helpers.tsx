import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import {ThemeProvider} from 'styled-components';
import {render as testRender} from '@testing-library/react';
import {theme} from '@flavorli/elements';

export function render(ui: React.ReactNode) {
  const utils = testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

  return utils;
}

export function renderWithRouter(ui: React.ReactElement, route: string = '/') {
  const history = createMemoryHistory({initialEntries: [route]});
  const Wrapper = ({children}: {children: React.ReactNode}) => (
    <Router history={history}>{children}</Router>
  );
  return {
    ...testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, {
      wrapper: Wrapper as React.ComponentType,
    }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history,
  };
}
