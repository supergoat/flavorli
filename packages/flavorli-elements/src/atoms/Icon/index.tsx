import React from 'react';
import {styled} from '../../theme';
import {icons} from './icons';

export interface IconProps {
  name: keyof typeof icons;
  size?: string;
  className?: string;
}

export default ({name, size, className}: IconProps) => {
  const icon = (icons[name] as unknown) as string;
  return <Icon src={icon} size={size} className={className} />;
};

export const Icon = styled.img<{size?: string}>`
  width: ${props => props.size};
  height: ${props => props.size};
`;
