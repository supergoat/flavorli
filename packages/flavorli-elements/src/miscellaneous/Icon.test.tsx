import React from 'react';
import {render} from '../test-helpers';
import {axe} from 'jest-axe';
import {Icon} from './Icon';

it('should not have any axe violations', async () => {
  const {container} = render(<Icon name="difficulty" />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

it('should display the icon that matches the icon name', () => {
  const {container} = render(<Icon name="difficulty" />);
  expect(container).toMatchSnapshot();
});

it('should change the icon size to the desired size', () => {
  const {container} = render(<Icon name="difficulty" size="40px" />);
  expect(container).toMatchSnapshot();
});
