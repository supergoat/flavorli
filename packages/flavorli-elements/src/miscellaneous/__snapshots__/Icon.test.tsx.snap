// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display the icon that matches the icon name 1`] = `
<img
  alt=""
  class="sc-fzXfLP czyQbP"
/>
`;

exports[`should change the icon size to the desired size 1`] = `
<img
  alt=""
  class="sc-fzXfLP eZxLPX"
/>
`;

exports[`should display the icon that matches the icon name 1`] = `
<img
  alt=""
  class="sc-fzXfLP czyQbP"
/>
`;

exports[`should display the icon with the correct size  1`] = `
<img
  alt=""
  class="sc-fzXfLP eZxLPX"
/>
`;

exports[`should have the correct styles 1`] = `
<img
  alt=""
  class="sc-fzXfLP czyQbP"
/>
`;
